etl/steps/data/meadow/war/2023-09-21/cow.py

```diff
@@ -33,63 +32,48 @@ def run(dest_dir: str) -> None:
-        tb = Table(df, short_name=snapshot_props["short_name"], underscore=True)
-        tb = tb.set_index(snapshot_props["index"], verify_integrity=True)
+        tb = snap.read(encoding=encoding, underscore=True)
+        # tb = Table(df, short_name=snapshot_props["short_name"], underscore=True)
+        tb = tb.set_index(snapshot_index, verify_integrity=True)
         tables.append(tb)
+
     # ZIP to table
-    ## Intra-state
-    snap = paths.load_snapshot("cow_intra_state.zip")
-    with tempfile.TemporaryDirectory() as tmpdir:
-        decompress_file(snap.path, tmpdir)
-        path = os.path.join(
-            tmpdir, "INTRA-STATE WARS v5.1 CSV.csv"
-        )  # other file: "INTRA-STATE_State_participants v5.1 CSV.csv"
-        encoding = infer_encoding(path)
-        log.info(f"cow: creating table from {snap.path}")
-        df = pd.read_csv(path, encoding=encoding)
-        tb = Table(df, short_name="intra_state", underscore=True)
-        tb = tb.set_index(["warnum"], verify_integrity=True)
-        tables.append(tb)
-    ## Inter-state (dydadic)
-    snap = paths.load_snapshot("cow_inter_state_dyadic.zip")
-    with tempfile.TemporaryDirectory() as tmpdir:
-        decompress_file(snap.path, tmpdir)
-        path = os.path.join(
-            tmpdir, "directed_dyadic_war.csv"
-        )  # other file: "INTRA-STATE_State_participants v5.1 CSV.csv"
-        encoding = infer_encoding(path)
-        log.info(f"cow: creating table from {snap.path}")
-        df = pd.read_csv(path, encoding=encoding)
-        tb = Table(df, short_name="inter_state_dyadic", underscore=True)
-        tb = tb.set_index(["warnum", "disno", "year", "statea", "stateb"], verify_integrity=True)
-        tables.append(tb)
+    snapshots = [
+        ("cow_intra_state.zip", "INTRA-STATE WARS v5.1 CSV.csv", ["warnum"]),
+        ("cow_inter_state_dyadic.zip", "directed_dyadic_war.csv", ["warnum", "disno", "year", "statea", "stateb"]),
+    ]
+    for uri, fname, index in snapshots:
+        snap = paths.load_snapshot(uri)
+        with tempfile.TemporaryDirectory() as tmpdir:
+            decompress_file(snap.path, tmpdir)
+            path = os.path.join(tmpdir, fname)  # other file: "INTRA-STATE_State_participants v5.1 CSV.csv"
+            encoding = infer_encoding(path)
+            log.info(f"cow: creating table from {snap.path}")
+            tb = pr.read_csv(
+                path, encoding=encoding, metadata=snap.to_table_metadata(), origin=snap.m.origin, underscore=True
+            )
+            tb = tb.set_index(index, verify_integrity=True)
+            tables.append(tb)
+
     #
     # Save outputs.
     #
     # Create a new meadow dataset with the same metadata as the snapshot.
     log.info("cow: creating dataset")
-    ds_meadow = create_dataset(dest_dir, tables=tables, default_metadata=snap.metadata)
+    ds_meadow = create_dataset(dest_dir, tables=tables, default_metadata=snap.metadata)  # type: ignore
```

Comment:
You could add the argument `check_variables_metadata=True`.

Reply:
for reference, Pablo meant this should be added in _all_ steps. This is something happening by default with `etl-wizard` templates, but this code is old so it is using bits from the old times.